services/src/main/java/org/keycloak/protocol/oidc/TokenManager.java

```diff
@@ -693,6 +694,29 @@ protected AccessToken initToken(RealmModel realm, ClientModel client, UserModel
         return token;
     }
 
+    private String getAcr(AuthenticatedClientSessionModel clientSession) {
+        int loa = Integer.parseInt(clientSession.getNote(Constants.LEVEL_OF_AUTHENTICATION));
+        if (loa < Constants.MINIMUM_LOA) {
+            loa = AuthenticationManager.isSSOAuthentication(clientSession) ? 0 : 1;
+        }
+
+        Map<String, Integer> acrLoaMap = AcrUtils.getAcrLoaMap(clientSession.getClient());
+        String acr = AcrUtils.mapLoaToAcr(loa, acrLoaMap, AcrUtils.getRequiredAcrValues(
+            clientSession.getNote(OIDCLoginProtocol.CLAIMS_PARAM)));
+        if (acr == null) {
+            acr = AcrUtils.mapLoaToAcr(loa, acrLoaMap, AcrUtils.getAcrValues(
+                clientSession.getNote(OIDCLoginProtocol.CLAIMS_PARAM),
+                clientSession.getNote(OIDCLoginProtocol.ACR_PARAM)));
+            if (acr == null) {
+                acr = AcrUtils.mapLoaToAcr(loa, acrLoaMap, acrLoaMap.keySet());
+                if (acr == null) {
+                    acr = String.valueOf(loa);
```

Comment:
[OIDC core specification](https://openid.net/specs/openid-connect-core-1_0.html#IDToken) mentions about acr:

> An absolute URI or an RFC 6711 [RFC6711] registered name SHOULD be used as the acr value;

We think that it would be more clear if we follow this suggestion.
As far as the implementation, we realm admin will be able to map loa value to a default URI. This default value will be returned to client if Keycloak can not map Loa to Acr.

Do you aggree?

Reply:
Implementation-wise, the logic here should be enough to support that as long as the admin has bound a LoA with a value, right?

W.r.t to the recommendation, we could make this more explicit in docs.

The logic here is also considering setting `0` in case the user is re-authenticating (cookie-based auth).

Reply:
Yes. If default value is not set, loa will be returned.
Default URI value for 0 can also be configured in the same way as other LOAs.